adapters/conversant/conversant.go

```diff
@@ -1,291 +1,177 @@
+		// These properties can also be specified in ad unit's video object,
+		// but are overridden if the custom params object also contains them.
 
-		if (imp.Secure == nil || *imp.Secure == 0) && params.Secure != nil {
-			imp.Secure = params.Secure
+		if len(cnvrExt.Protocols) > 0 {
+			imp.Video.Protocols = make([]openrtb.Protocol, 0, len(cnvrExt.Protocols))
+			for _, protocol := range cnvrExt.Protocols {
+				imp.Video.Protocols = append(imp.Video.Protocols, openrtb.Protocol(protocol))
+			}
 		}
-	}
-
-	// Do a quick check on required parameters
 
-	if cnvrReq.Site != nil && cnvrReq.Site.ID == "" {
-		return nil, &errortypes.BadInput{
-			Message: "Missing site id",
+		if len(cnvrExt.MIMEs) > 0 {
+			imp.Video.MIMEs = make([]string, len(cnvrExt.MIMEs))
+			copy(imp.Video.MIMEs, cnvrExt.MIMEs)
 		}
-	}
 
-	if cnvrReq.App != nil && cnvrReq.App.ID == "" {
-		return nil, &errortypes.BadInput{
-			Message: "Missing app id",
+		if cnvrExt.MaxDuration != nil {
+			imp.Video.MaxDuration = *cnvrExt.MaxDuration
 		}
 	}
+}
 
-	// Start capturing debug info
-
-	debug := &pbs.BidderDebug{
-		RequestURI: a.URI,
-	}
-
-	if cnvrReq.Device == nil {
-		cnvrReq.Device = &openrtb.Device{}
-	}
-
-	// Convert request to json to be sent over http
-
-	j, _ := json.Marshal(cnvrReq)
-
-	if req.IsDebug {
-		debug.RequestBody = string(j)
-		bidder.Debug = append(bidder.Debug, debug)
-	}
-
-	httpReq, err := http.NewRequest("POST", a.URI, bytes.NewBuffer(j))
-	httpReq.Header.Add("Content-Type", "application/json")
-	httpReq.Header.Add("Accept", "application/json")
-
-	resp, err := ctxhttp.Do(ctx, a.http.Client, httpReq)
-	if err != nil {
-		return nil, err
-	}
-
-	defer resp.Body.Close()
-
-	if req.IsDebug {
-		debug.StatusCode = resp.StatusCode
-	}
-
-	if resp.StatusCode == 204 {
-		return nil, nil
+func (c ConversantAdapter) MakeBids(internalRequest *openrtb.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	var errs []error
+	if response.StatusCode == http.StatusNoContent {
+		return nil, nil // no bid response
 	}
 
-	body, err := ioutil.ReadAll(resp.Body)
-
-	if err != nil {
-		return nil, err
+	if response.StatusCode != http.StatusOK {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Unexpected status code: %d", response.StatusCode),
+		}}
 	}
 
-	if resp.StatusCode == http.StatusBadRequest {
-		return nil, &errortypes.BadInput{
-			Message: fmt.Sprintf("HTTP status: %d, body: %s", resp.StatusCode, string(body)),
-		}
+	var resp openrtb.BidResponse
+	if err := json.Unmarshal(response.Body, &resp); err != nil {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("bad server response: %d. ", err),
+		}}
 	}
 
-	if resp.StatusCode != 200 {
-		return nil, &errortypes.BadServerResponse{
-			Message: fmt.Sprintf("HTTP status: %d, body: %s", resp.StatusCode, string(body)),
-		}
+	if resp.SeatBid == nil {
```

Comment:
This should be `len(resp.SeatBid) == 0` to protect from safely accessing `resp.SeatBid[0].Bid` below. The SeatBid slice could be non-nil, but still empty.